### PR TITLE
Additivefoam input file writer & test

### PR DESCRIPTION
**Pull request by hespuck19**
Does not account for the fluid flow properties typically found in transportProperties file.

### CODE REVIEW

File: mistlib/core.py
Comment (stvdwtt) [inline mistlib/core.py:432]: I strongly recommend using descriptive names whenever possible. Using something like `f` for a file or `p` for a property can be ok since the letter stands for something, but even that is best used when all uses are in close proximity.

```suggestion
        thermal_conductivity_coeffs = []
```

File: mistlib/core.py
Comment (stvdwtt) [inline mistlib/core.py:474]: Is there a limitation in additiveFOAM that thermal conductivity has just two terms and the specific heat has just one term? Or is that just what Gerry had in the example? If the additiveFOAM input generally accepts 3 polynomial coefficients, then we should be able to supply as many as they can use.

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:407]: Multi-line string formatting issue? We could add in black or yapf to auto-format

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:404]: Is this good python practice? I'm a little out of practice

**Comment by stvdwtt** [comment]
@hespuck19 -- I haven't had a chance to look at the latest commits in detail, but in case you didn't see it yet, the CI failed for your new test. (It looks like the function is expecting a path as input, but isn't getting one or maybe is getting an empty string).

File: mistlib/core.py
Comment (gknapp1) [reply mistlib/core.py]: AdditiveFOAM uses the [OpenFOAM Polynomial class](https://cpp.openfoam.org/v10/classFoam_1_1Polynomial.html) here, which can accept N-order polynomials. Practically, I don't anticipate we would use more than a third order polynomial though.

File: mistlib/core.py
Comment (streeve) [reply mistlib/core.py:404]: Opened #19

**Comment by stvdwtt** [comment]
@hespuck19, do you have a preferred order to merge this versus #20 or #21? I see that #21 builds off of this branch.

Options that make sense to me:
1) Finish fixing up this PR and merge it, then merge #20 or #21, then do a new PR to use `get_property` for the additiveFOAM writer.
2) Fix the test in #20 and merge it, then use `get_property` in this PR and merge it.
3) Fix the test in #21 and merge it, close this PR without merging (since #21) has the commits, then do a new PR to use `get_property` for the additiveFOAM writer.

File: mistlib/core.py
Comment (stvdwtt) [reply mistlib/core.py]: @gknapp1, my understanding from @colemanjs is that additiveFOAM is hard-coded to only accept up to third-order polynomials, even if the OpenFOAM class is more general.

@hespuck19, I'm guessing it's on your to-do list and you just haven't gotten to it yet, but I think the main remaining thing to change for this PR is to generalize all of the parameters so that they can be up to third-order polynomials. Although maybe you want `get_property` working before you do that.

Comment (gknapp1) [reply mistlib/core.py]: @stvdwtt you're correct. That's my mistake, I missed the hard-coded polynomial size when I checked the code. 

Though I think it is what you meant, to clarify, AdditiveFOAM is hard-coded to accept 3 polynomial *coefficients* (exponents of 0, 1, and 2), so it would technically only accept a second-order polynomial.

Comment (stvdwtt) [reply mistlib/core.py]: @gknapp1, thanks for that clarification, I must have switched three coefficients/third order polynomial in my notes from talking to John.

**Comment by streeve** [comment]
> @hespuck19, do you have a preferred order to merge this versus #20 or #21? I see that #21 builds off of this branch.
> 
I opened #22 to clarify that there's an additional failure (that we may want to ignore in order to merge) unrelated to both

Whatever we decide to do, each pull request should address one feature/issue and we can "force" push any branch to get it to a state that we want

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:388]: Maybe remove?

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:438]: It's most performant to create a long string, then write once (probably not a big deal here)

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:445]: Looks like another good helper function (additivefoam specific this time though)

File: mistlib/core.py
Comment (streeve) [inline mistlib/core.py:491]: Should there be a top-level function `write_additivefoam_input` that calls all the other ones?

Comment (stvdwtt) [reply mistlib/core.py:507]: I think it would make sense to have a top-level function to do everything you need for additivefoam.

Comment (hespuck19) [reply mistlib/core.py:507]: Working on the top level function for AdditiveFOAM and a better way to handle polynomial values for material constants.

**Comment by stvdwtt** [comment]
Talking to @hespuck19, here's what we want before merging:
- Add a top level function to create all AdditiveFOAM inputs (see discussion above)
- Add a new helper function to get a zero-padded array of coefficients for a property (that can handle either a Laurent polynomial or a scalar property)

File: mistlib/core.py
Comment (streeve) [reply mistlib/core.py:419]: Added

File: mistlib/core.py
Comment (streeve) [reply mistlib/core.py]: Added

**Comment by streeve** [comment]
@stvdwtt I think I got this mostly working @gknapp1 honestly not sure if this actually works with additivefoam yet

File: mistlib/core.py
Comment (stvdwtt) [inline mistlib/core.py:410]: @gknapp1 -- can any of these be polynomials or are they all scalars?

**Review by stvdwtt — APPROVED** [review]
@gknapp1, I'll leave it to you to see if you're satisfied that this will work for additiveFOAM, but I'm happy with this now. Thanks @streeve!

**Review by gknapp1 — APPROVED** [review]

**Pull request merged**
